src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClient.java

```diff
@@ -213,60 +205,29 @@ private String getRequest(String path) throws StashApiException {
     // address large amount of close_wait sockets client and fin sockets server side
     request.addHeader("Connection", "close");
 
-    String response = null;
-    FutureTask<String> httpTask = null;
-    Thread thread;
+    String response = "";
     try {
       request.addHeader(new BasicScheme().authenticate(credentials, request, null));
+      HttpResponse httpResponse = client.execute(request);
```

Comment:
it seems this block is mostly the same for `get` an `post`, is there a chance to do some code reuse?

Reply:
Yes, but not in this PR. My first priority is correctness. Code reuse will be done once it's clear which code we need. For instance, POST requests can look at the HTTP status code code or at the response contents or at both. I will need to decide what is best by looking at the API specification. Then I will see if there is a pattern that can be refactored.

I may actually need to go in the opposite direction at one point. `validResponseCode` is misleading, as different requests expect different status codes. `409 Conflict` is unquestionably valid for the merge requests. `validResponseCode` conflates two issues, valid/invalid and success/error.

Reply:
Actually, the HttpResponse contents processing can be refactored. I did it as the first commit, which reduced the amount of code being moved in the larger second commit. I also decided to throw on empty response contents, as we expect JSON, and an empty string is not valid JSON anyway.